app/src/main/java/ru/netology/fmhandroid/adapter/NoteListAdapter.kt

```diff
@@ -0,0 +1,95 @@
+import ru.netology.fmhandroid.dto.Note
+import ru.netology.fmhandroid.enum.ExecutionPriority
+import ru.netology.fmhandroid.utils.Utils
+import java.time.LocalDateTime
+
+interface OnNoteItemClickListener {
+    fun onCard(note: Note) {}
+}
+
+class NoteListAdapter(
+    private val onNoteItemClickListener: OnNoteItemClickListener
+) : ListAdapter<Note, NoteListAdapter.NoteViewHolder>(NoteDiffCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
+        val binding = NoteCardItemForListBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return NoteViewHolder(binding, onNoteItemClickListener)
+    }
+
+    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
+        getItem(position)?.let {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                holder.bind(it)
+            }
+        }
+    }
+
+    class NoteViewHolder(
+        private val binding: NoteCardItemForListBinding,
+        private val onNoteItemClickListener: OnNoteItemClickListener
+    ) : RecyclerView.ViewHolder(binding.root) {
+        @RequiresApi(Build.VERSION_CODES.O)
+        fun bind(note: Note) {
+
+            binding.apply {
+                prioritization(note)
+                noteCardItemForListPatientName.text = note.shortPatientName
+                noteCardItemForListExecutorName.text = note.shortExecutorName
+                noteCardItemForListPlanDate.text = Utils.convertDate(note.planeExecuteDate!!)
+
+                if (note.factExecuteDate == null) {
+                    noteCardItemForListFactDate.setText(R.string.date_not_set)
+                } else {
+                    noteCardItemForListFactDate.text = Utils.convertDate(note.factExecuteDate)
+                }
+                noteCardItemForListDescription.text = note.description
+                noteCardItemForListDescription.setOnClickListener {
+                    onNoteItemClickListener.onCard(note)
+                }
+            }
+        }
+
+        private fun NoteCardItemForListBinding.prioritization(note: Note) {
+            val executionPriority = note.planeExecuteDate?.let {
+                BusinessRules.determiningPriorityLevelOfNote(
+                    LocalDateTime.now(),
+                    it
+                )
+            }
+            when (executionPriority) {
+                ExecutionPriority.HIGH ->
+                    this.noteCardItemForListPriorityIndicator.setImageResource(R.color.red)
+                ExecutionPriority.MEDIUM ->
+                    this.noteCardItemForListPriorityIndicator.setImageResource(R.color.yellow)
+                ExecutionPriority.LOW ->
+                    this.noteCardItemForListPriorityIndicator.setImageResource(R.color.green)
+            }
+        }
+    }
+
+    class NoteDiffCallback : DiffUtil.ItemCallback<Note>() {
```

Comment:
```suggestion
    private object NoteDiffCallback : DiffUtil.ItemCallback<Note>() {
```
В итоге уменьшится видимость класса и не надо будет вызывать конструктор в месте использования

Reply:
Исправлено